pkg/i2gw/providers/ingressnginx/ingressnginx.go

```diff
@@ -45,15 +46,21 @@ func NewProvider(conf *i2gw.ProviderConf) i2gw.Provider {
 		storage:        newResourcesStorage(),
 		resourceReader: newResourceReader(conf),
 		converter:      newConverter(),
+		extConverter:   newExtensionConverter(),
 	}
 }
 
-// ToGatewayAPI converts stored Ingress-Nginx API entities to i2gw.GatewayResources
+// ToIR converts stored Ingress-Nginx API entities to i2gw.IR
 // including the ingress-nginx specific features.
-func (p *Provider) ToGatewayAPI() (i2gw.GatewayResources, field.ErrorList) {
+func (p *Provider) ToIR() (i2gw.IR, field.ErrorList) {
 	return p.converter.convert(p.storage)
 }
 
+func (p *Provider) ToGatewayAPIAndExtensions(ir i2gw.IR) (i2gw.GatewayResources, field.ErrorList) {
+	return p.extConverter.irToGateway(ir)
+
+}
```

Comment:
we may probably change it based on my previous comments on the interface, but why not simply this? 

```go
func (p *Provider) ToGatewayAPIAndExtensions(ir i2gw.IR) (i2gw.GatewayResources, field.ErrorList) {
	return common.ToGatewayResources(ir)
}
```

Reply:
Sounds good, I'll update this for other providers as well. They can use the common function until they want to create a dedicate object to include translation logics for extensions.